### PR TITLE
feat(button): Add blue css buttons

### PR DESCRIPTION
**Pull request by mannycarrera4**
## Summary

Added css for blue buttons, renamed variables and class names for orange buttons to deprecated.

### Code Changes
- Added css for the new `blue primary, secondary, delete, and dropdown` buttons
- Added `button-deprecated.scss` for the old buttons
- Added `button-deprecated-variables.scss` for the old buttons variables
- Added `accessibility-deprecated.scss` for the old buttons accessibility
- Added `icon-button-variables.scss` for icon button variables
- For the deprecated buttons, added `wdc-btn-deprecated` class name
- `wdc-btn` now styles the blue buttons
- A deprecation warning will show on compilation if you use the `wdc-btn-deprecated` mixin.

### Breaking Changes
- Moved old style buttons to `button-deprecated.scss`. You now need to use `wdc-btn-deprecated` to maintain these styles. Using `wdc-btn` will now result in the new style of buttons
- Moved old style button variables to `button-deprecated-variables.scss`
- Moved old style button accessibility styles to `accessibility-deprecated.scss`

## Checklist

- [x] branch has been rebased on the latest master commit
- [x] code has been documented and, if applicable, usage described in README.md
- [x] design approved final implementation
- [x] a11y approved final implementation
- [x] code adheres to the [API & Pattern guidelines](../API_PATTERN_GUIDELINES.md)

### CODE REVIEW

File: modules/button/css/lib/button.scss [hunk 1]
```diff
@@ -7,104 +7,53 @@ $_wdc-btn-caret-down-icon-hover: svg-load(
...
+  padding: 0 14px;
+  background-color: $wdc-btn-bg-secondary;
+  color: $wdc-btn-secondary-text-color;
+  border-radius: 999px;
+  transition: 0.2s background, 0.2s border-color, 0.2s box-shadow;
   text-align: center;
   font-size: 14px;
   cursor: pointer;
   box-sizing: border-box;
-  -webkit-font-smoothing: antialiased;
```
Comment (anicholls) [inline modules/button/css/lib/button.scss]: Can you add this back in? We use antialiasing on the React buttons

File: modules/button/css/lib/button.scss [hunk 2]
```diff
@@ -7,104 +7,53 @@ $_wdc-btn-caret-down-icon-hover: svg-load(
...
+  min-width: 96px;
   max-width: $wdc-btn-max-width-l;
   height: $wdc-btn-height-l;
-  padding: $wdc-btn-padding-l;
-  background-color: $wdc-btn-bg;
-  border: 1px solid $wdc-btn-border;
-  color: $wdc-btn-color;
-  border-radius: $wdc-btn-border-radius;
-  transition: all 120ms, box-shadow 100ms;
+  padding: 0 14px;
```
Comment (anicholls) [inline modules/button/css/lib/button.scss]: This doesn't account for the nested span with 8px of padding in the react buttons. This should be `24px`. The React version is actually 2px off. Can you fix this in #191 (Large should be 24 + 8, Medium should be 16 + 8 and Small should be 16 https://github.com/Workday/canvas-kit/blob/master/modules/button/react/lib/ButtonStyles.ts#L232)?

File: modules/button/css/lib/button.scss [hunk 3]
```diff
@@ -7,104 +7,53 @@ $_wdc-btn-caret-down-icon-hover: svg-load(
...
+  &:hover {
+    background-color: $wdc-btn-bg-secondary-hover;
   }
 
-  .wd-icon-accent,
-  .wd-icon-fill {
-    fill: $wdc-btn-icon-color;
+  &.wdc-btn-active,
+  &:active {
+    background-color: $wdc-btn-bg-secondary-active;
```
Comment (anicholls) [inline modules/button/css/lib/button.scss]: The focus styles are overriding this, so pressing the spacebar while focussed doesn't show any background color change. This is the case for all of the new buttons

File: modules/button/css/lib/button.scss [hunk 4]
```diff
@@ -7,104 +7,53 @@ $_wdc-btn-caret-down-icon-hover: svg-load(
...
-  padding: $wdc-btn-padding-l;
-  background-color: $wdc-btn-bg;
-  border: 1px solid $wdc-btn-border;
-  color: $wdc-btn-color;
-  border-radius: $wdc-btn-border-radius;
-  transition: all 120ms, box-shadow 100ms;
+  padding: 0 14px;
+  background-color: $wdc-btn-bg-secondary;
+  color: $wdc-btn-secondary-text-color;
+  border-radius: 999px;
```
Comment (anicholls) [inline modules/button/css/lib/button.scss]: Nit: `$wdc-border-radius-circle`

File: modules/button/css/lib/button.scss [hunk 5]
```diff
@@ -133,246 +84,94 @@ $_wdc-btn-caret-down-icon-hover: svg-load(
...
+    &.wdc-btn-hover:not([disabled]):not(.wdc-btn-disabled):after,
+    &:hover:not([disabled]):not(.wdc-btn-disabled):after {
+      background-image: $_wdc-btn-caret-down-icon-white;
     }
   }
 
+  &.wdc-btn-size-l {
+    min-width: 112px;
+    height: 48px;
+    padding: 0 18px;
```
Comment (anicholls) [inline modules/button/css/lib/button.scss]: This should be 24px

File: modules/button/css/lib/variables.scss [hunk 6]
```diff
@@ -1,94 +1,30 @@
...
+$wdc-btn-bg-primary-disabled: $wdc-color-blueberry-200;
+
+$wdc-btn-bg-delete: $wdc-color-cinnamon-500;
+$wdc-btn-bg-delete-focus: $wdc-color-cinnamon-500;
+$wdc-btn-bg-delete-disabled: $wdc-color-cinnamon-200;
+$wdc-btn-bg-delete-hover: $wdc-color-cinnamon-600;
+$wdc-btn-bg-delete-active: #80160e;
 
-$wdc-btn-font-weight: 500;
 $wdc-btn-border-size: 1px;
```
Comment (anicholls) [inline modules/button/css/lib/variables.scss]: Not used

File: modules/button/css/lib/variables.scss [hunk 7]
```diff
@@ -1,94 +1,30 @@
...
+$wdc-btn-border-color-focus: $wdc-color-blueberry-500;
```
Comment (anicholls) [inline modules/button/css/lib/variables.scss]: Not used

File: modules/button/css/lib/button.scss [hunk 8]
```diff
@@ -7,104 +7,53 @@ $_wdc-btn-caret-down-icon-hover: svg-load(
...
-  background-color: $wdc-btn-bg;
-  border: 1px solid $wdc-btn-border;
-  color: $wdc-btn-color;
-  border-radius: $wdc-btn-border-radius;
-  transition: all 120ms, box-shadow 100ms;
+  padding: 0 14px;
+  background-color: $wdc-btn-bg-secondary;
+  color: $wdc-btn-secondary-text-color;
+  border-radius: 999px;
+  transition: 0.2s background, 0.2s border-color, 0.2s box-shadow;
```
Comment (anicholls) [inline modules/button/css/lib/button.scss]: These should all be `120ms`

File: modules/button/css/lib/button-deprecated.scss [hunk 9]
```diff
@@ -0,0 +1,386 @@
+$_wdc-btn-deprecated-caret-down-icon: svg-load(
+  '#{$_wdc-btn-icons-path}/wd-icon-caret-down.svg',
+  fill=$wdc-btn-icon-color
+);
+$_wdc-btn-deprecated-caret-down-icon-hover: svg-load(
+  '#{$_wdc-btn-icons-path}/wd-icon-caret-down.svg',
+  fill=$wdc-btn-icon-hover-color
+);
+
+@mixin wdc-btn-icon-inline($_icon) {
```
Comment (anicholls) [inline modules/button/css/lib/button-deprecated.scss]: This is still used in the new buttons for dropdown buttons. If this file gets deleted when we complete the deprecation we'll have issues

File: modules/button/css/README.md [hunk 10]
```diff
@@ -25,6 +25,26 @@ Add your `node_modules` directory to your SASS `includePaths`. You will then be
 [postcss-inline-svg](https://github.com/TrySound/postcss-inline-svg) plugin to properly process and
 inline icons. Process your SASS through PostCSS once it has been compiled to CSS.
 
+## Deprecation Warning
+
+> We are rolling out our next iteration of our buttons and will be deprecating the old ones(orange
+> primary, secondary, and delete) on (this time frame).
+
+> We've renamed the old buttons to use `wdc-button-deprecated` mixin and have added a warning
```
Comment (anicholls) [inline modules/button/css/README.md]: Most people don't use the mixins for components and there's no warning associated with using the class `wdc-btn-deprecated`. I would just remove this messaging.

File: modules/button/css/README.md [hunk 11]
```diff
@@ -25,6 +25,26 @@ Add your `node_modules` directory to your SASS `includePaths`. You will then be
...
+
+### New Button
+
+You will be opt in to the new buttons with (insert version number) without having to change your
+class names. The new buttons include: blue primary button, secondary, delete, icon buttons and
+dropdown.
+
+### Deprecated Buttons
+
+The old buttons (orange primary, secondary, delete, and split) are still available, but you'll have
```
Comment (anicholls) [inline modules/button/css/README.md]: Can you provide an example of this?

File: modules/button/css/lib/button.scss (hunk 3 above)
Comment (mannycarrera4) [reply modules/button/css/lib/button.scss]: @anicholls i added this to accessibility.scss, do I still need these styles here? Can you double check, please, thanks!

File: modules/button/css/lib/variables.scss [hunk 12]
```diff
@@ -1,94 +1,30 @@
...
-$wdc-btn-icon-disabled-color: $wdc-color-soap-600;
-$wdc-btn-icon-inverse-color: $wdc-color-french-vanilla-100;
-$wdc-btn-icon-inverse-disabled-color: rgba(255, 255, 255, 0.75);
+$wdc-btn-height-l: $wdc-spacing-xl;
 
-$wdc-btn-icon-width: $wdc-spacing-xl;
-$wdc-btn-icon-height: $wdc-spacing-xl;
-$wdc-btn-icon-size-s: $wdc-spacing-s;
-$wdc-btn-icon-size-m: $wdc-spacing-m;
+$wdc-border-radius-circle: 999px;
```
Comment (anicholls) [inline modules/button/css/lib/variables.scss]: This is already available via #204

Comment (mannycarrera4) [reply modules/button/css/lib/variables.scss]: neat! will update

File: modules/button/css/lib/button.scss (hunk 3 above)
Comment (anicholls) [reply modules/button/css/lib/button.scss]: Looks good 👍

**Pull request merged**